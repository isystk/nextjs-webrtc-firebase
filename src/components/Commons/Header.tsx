import React, { FC } from 'react'
import { AppBar, Toolbar, IconButton } from '@material-ui/core'
import MenuIcon from '@material-ui/icons/Menu'
import { MenuContext, MenuContextProps } from '@/components/Home'

export const CommonHeader: FC = () => {
  const DEAULT_TITLE = process.env.APP_NAME
  return (
    <MenuContext.Consumer>
      {(value: MenuContextProps) => {
        const { isMenuOpen, setMenuOpen, roomName } = value
        return (
          <>
            <AppBar position="fixed" className="App-header">
              <Toolbar>
                <IconButton
                  color="inherit"
                  disabled={!roomName}
                  onClick={() => setMenuOpen(!isMenuOpen)}
                >
                  <MenuIcon />
                </IconButton>
                <div className="App-logo">{roomName || DEAULT_TITLE}</div>
              </Toolbar>
            </AppBar>
          </>
        )
      }}
    </MenuContext.Consumer>
  )
}

export default CommonHeader
