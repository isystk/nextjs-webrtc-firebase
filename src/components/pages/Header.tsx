import React, { Dispatch, SetStateAction, VFC } from 'react'
import { AppBar, Toolbar, IconButton } from '@material-ui/core'
import MenuIcon from '@material-ui/icons/Menu'
import { useSelector } from 'react-redux'
import { Client } from '@/store/StoreTypes'

type Props = {
  isMenuOpen: boolean
  setMenuOpen: Dispatch<SetStateAction<boolean>>
  client: Client
}

const CommonHeader: VFC<Props> = ({ isMenuOpen, setMenuOpen }) => {
  const { client } = useSelector((state: {client: Client}) => state.client)
  const DEAULT_TITLE = process.env.APP_NAME
  return (
    <>
      <AppBar position="fixed" className="App-header">
        <Toolbar>
          <IconButton
            color="inherit"
            disabled={!client.roomName}
            onClick={() => setMenuOpen(!isMenuOpen)}
          >
            <MenuIcon />
          </IconButton>
          <div className="App-logo">{client.roomName || DEAULT_TITLE}</div>
        </Toolbar>
      </AppBar>
    </>
  )
}

export default CommonHeader
