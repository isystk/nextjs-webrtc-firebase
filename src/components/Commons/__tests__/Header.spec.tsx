import React from 'react'
import renderer from 'react-test-renderer'
import { CommonHeader } from '../Header'

test('Header', () => {
  const component = renderer.create(<CommonHeader />)
  const tree = component.toJSON()

  expect(tree).toMatchSnapshot()
})
