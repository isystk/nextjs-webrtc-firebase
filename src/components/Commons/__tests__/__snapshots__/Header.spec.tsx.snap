// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 1`] = `
<nav
  className="main-header navbar navbar-expand navbar-white navbar-light"
>
  <ul
    className="navbar-nav"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        data-widget="pushmenu"
        href="#"
      >
        <i
          className="fas fa-bars"
        />
      </a>
    </li>
  </ul>
</nav>
`;
